compiler/noirc_evaluator/src/ssa/ir/instruction.rs

```diff
@@ -112,6 +112,9 @@ impl Intrinsic {
     /// Returns whether the `Intrinsic` has side effects.
     ///
     /// If there are no side effects then the `Intrinsic` can be removed if the result is unused.
+    ///
+    /// An obvious example of a side effect is `println`, or pushing to a slice, but functions
+    /// which can fail due to implicit constraints are also considered to have a side effect.
```

Comment:
The `println` example doesn't apply here as that's a foreign call rather than an intrinsic.

Reply:
I used it because [it's still in the docs](https://github.com/noir-lang/noir/blob/7d612425728a9d5852ec26a96af629aec7cd0a21/compiler/noirc_evaluator/src/ssa/ir/instruction.rs#L48-L49) of `Intrinsic` and it also appeared in [this comment](https://github.com/noir-lang/noir/issues/6533#issuecomment-2479731097), so I thought it might be some kind of honorary intrinsic, or a vestigial one. 

I'm happy to remove it from both places to avoid the confusion.